Tiltfile

```diff
@@ -3,82 +3,101 @@
-  img.add(repo.path('./%s/yarn.lock' % name), '/app/yarn.lock')
-  img.add(repo.path('./%s/' % name), "/app")
+  img.add(repo.path('snack/'), path)
 
-  if run_fn:
-    run_fn(img)
+  img.run('go install github.com/windmilleng/servantes/snack')
 
   return k8s_service(yaml, img)
 
-def fe():
-  return go_service('servantes')
+def doggos():
+  yaml = read_file('doggos/deployments/doggos.yaml')
 
-def vigoda():
-  return go_service('vigoda')
+  image_name = 'gcr.io/windmill-public-containers/servantes/doggos'
 
-def snack():
-  return go_service('snack')
+  img = build_docker_image('Dockerfile.go.base', image_name, '/go/bin/doggos')
+  path = '/go/src/github.com/windmilleng/servantes/doggos'
+  repo = local_git_repo('.')
+  img.add(repo.path('doggos/'), path)
 
-def doggos():
-  return go_service('doggos')
+  img.run('go install github.com/windmilleng/servantes/doggos')
+
+  return k8s_service(yaml, img)
 
 def fortune():
-  def runs(img):
-    img.run('cd src/github.com/windmilleng/servantes/fortune && make proto')
+  yaml = read_file('fortune/deployments/fortune.yaml')
+
+  image_name = 'gcr.io/windmill-public-containers/servantes/fortune'
 
-  return go_service('fortune', runs)
+  img = build_docker_image('Dockerfile.go.base', image_name, '/go/bin/fortune')
+  path = '/go/src/github.com/windmilleng/servantes/fortune'
+  repo = local_git_repo('.')
+  img.add(repo.path('fortune/'), path)
+
+  img.run('cd src/github.com/windmilleng/servantes/fortune && make proto')
+  img.run('go install github.com/windmilleng/servantes/fortune')
+
+  return k8s_service(yaml, img)
 
 def hypothesizer():
-  def runs(img):
-    img.run('cd /app && pip install -r requirements.txt', trigger='hypothesizer/requirements.txt')
+  yaml = read_file('hypothesizer/deployments/hypothesizer.yaml')
+
+  image_name = 'gcr.io/windmill-public-containers/servantes/hypothesizer'
+
+  img = build_docker_image('Dockerfile.py.base', image_name, 'python /app/app.py')
+  repo = local_git_repo('.')
+  img.add(repo.path('hypothesizer/'), "/app")
 
-  return python_service('hypothesizer', runs)
+  img.run('cd /app && pip install -r requirements.txt', trigger='hypothesizer/requirements.txt')
+
+  return k8s_service(yaml, img)
 
 def spoonerisms():
-  def runs(img):
-    img.run('cd /app && yarn install', trigger=['spoonerisms/package.json', 'spoonerisms/yarn.lock'])
+  yaml = read_file('spoonerisms/deployments/spoonerisms.yaml')
 
-  return javascript_service('spoonerisms', 'src', runs)
+  image_name = 'gcr.io/windmill-public-containers/servantes/spoonerisms'
 
-def local_file(p):
-  return local("cat %s" % p)
+  img = build_docker_image('Dockerfile.js.base', image_name, 'node /app/index.js')
+  repo = local_git_repo('.')
+  img.add(repo.path('spoonerisms/src'), '/app/')
+  img.add(repo.path('spoonerisms/package.json'), '/app/package.json')
+  img.add(repo.path('spoonerisms/yarn.lock'), '/app/yarn.lock')
+  img.add(repo.path('spoonerisms/'), "/app")
```

Comment:
???

are you sure we want this? this seems to make the previous 3 lines redundant

Reply:
oh whoops, yeah I don't want that last line.